idutils/ext.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of IDUtils
+# Copyright (C) 2024 CERN.
+#
+# IDUtils is free software; you can redistribute it and/or modify
+# it under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Invenio IDUtils module for managing persistent identifiers used in scholarly communication."""
+
+from importlib_metadata import entry_points
+
+
+class IDUtils(object):
+    """Invenio extension."""
+
+    def __init__(self, app=None):
+        """Extension initialization."""
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app):
+        """Flask application initialiation."""
+        self.init_idutils_registry()
+        app.extensions["idutils"] = self
+
+    def init_idutils_registry(self):
+        """Initialize identifier registries."""
+        # For registering is_* checks (functions)
+        self.identifier_checks_registry = {}
+        self.register_entry_point(
```

Comment:
minor: @sakshamarora1 was this a convention you followed? If not, what about:
```suggestion
        self._load_entry_point(
```